pandas/core/internals/blocks.py

```diff
@@ -1005,8 +1006,14 @@ def coerce_to_target_dtype(self, other):
         we can also safely try to coerce to the same dtype
         and will receive the same block
         """
-        # if we cannot then coerce to object
-        dtype, _ = infer_dtype_from(other, pandas_dtype=True)
+        if isinstance(other, str):
+            if is_string_dtype(self.dtype):
+                return self
+            else:
+                dtype = np.object_
+        else:
+            # if we cannot then coerce to object
+            dtype, _ = infer_dtype_from(other, pandas_dtype=True)
```

Comment:
The suggestion from @jreback [here](https://github.com/pandas-dev/pandas/pull/33622#discussion_r412295758) was to make `infer_dtype_from` return `StringDtype` when appropriate, so that L1090-L1013 here wouldnt be necessary

Reply:
I tried to move into infer_dtype_from(), but a lot of tests fail. So, I didn't move.